pkg/tools/etcd_helper_watch.go

```diff
@@ -250,6 +259,10 @@ func (w *etcdWatcher) translate() {
 			return
 		case res, ok := <-w.etcdIncoming:
 			if ok {
+				if curLen := len(w.etcdIncoming); curLen > 0 {
```

Comment:
This is going to cause a logging storm, which will further exacerbate the backlog. Don't you rather want this logging threshold to be greater than zero?  Perhaps n/2 for a channel of length n?  And you don't want to log more often than once a minute or so.